harness/determined/common/experimental/trial.py

```diff
@@ -165,59 +196,84 @@ def select_checkpoint(
 
-        def get_one(offset: int) -> bindings.v1GetTrialCheckpointsResponse:
+        order_by = None
+        if latest:
+            sort_by = CheckpointSortBy.SORT_BY_BATCH_NUMBER
+            order_by = CheckpointOrderBy.DESC
+
+        if sort_by:
+            order_by = CheckpointOrderBy.ASC if smaller_is_better else CheckpointOrderBy.DESC
+
+        checkpoints = self.get_checkpoints(sort_by=sort_by, order_by=order_by)
+
+        return checkpoints[0] if checkpoints else []
+
+    def get_checkpoints(
+        self,
+        sort_by: Optional[Union[str, CheckpointSortBy]] = None,
+        order_by: Optional[CheckpointOrderBy] = None,
+    ) -> List[checkpoint.Checkpoint]:
+        """
+        Return a list of :class:`~determined.experimental.Checkpoint` instances for the current
+        trial.
+
+        Either ``sort_by`` and ``order_by`` are both specified or neither are.
+
+        Arguments:
+            sort_by (string, :class:`~determined.experimental.CheckpointSortBy`): Which field to
+                sort by. Strings are assumed to be searcher metric names.
+            order_by (:class:`~determined.experimental.CheckpointOrderBy`): Whether to sort in
+                ascending or descending order.
+        """
+
+        if (sort_by is None) != (order_by is None):
+            raise AssertionError("sort_by and order_by must be set together")
+
+        def get_trial_checkpoints(offset: int) -> bindings.v1GetTrialCheckpointsResponse:
             return bindings.get_GetTrialCheckpoints(
                 self._session,
                 id=self.id,
-                orderBy=bindings.v1OrderBy.ORDER_BY_DESC,
-                sortBy=bindings.v1GetTrialCheckpointsRequestSortBy.SORT_BY_BATCH_NUMBER,
+                orderBy=order_by and order_by._to_bindings(),
+                sortBy=sort_by._to_bindings() if isinstance(sort_by, CheckpointSortBy) else None,
                 offset=offset,
             )
 
-        resps = api.read_paginated(get_one)
+        resps = api.read_paginated(get_trial_checkpoints)
 
         checkpoints = [
             checkpoint.Checkpoint._from_bindings(c, self._session)
             for r in resps
             for c in r.checkpoints
         ]
 
-        if not checkpoints:
-            raise AssertionError("No checkpoint found for trial {}".format(self.id))
-
-        if latest:
-            return checkpoints[0]
+        # If sort_by was a defined field, we already sorted and ordered.
+        if isinstance(sort_by, CheckpointSortBy) or not checkpoints:
+            return checkpoints
 
+        # If sort not specified, sort and order default to searcher configs.
         if not sort_by:
             training = checkpoints[0].training
             assert training
             config = training.experiment_config
-            sb = config.get("searcher", {}).get("metric")
-            if not isinstance(sb, str):
+            searcher_metric = config.get("searcher", {}).get("metric")
```

Comment:
*question*: why are there two `get` calls as in  `config.get("searcher", {}).get("metric")` and below? If `"searcher"` isn't a key, doesn't the second `get` error?

Reply:
see python docs for .get: https://docs.python.org/3/library/stdtypes.html#dict.get

basically if "searcher" isn't a key, get returns the default which is {} instead of throwing KeyError. if you didn't use get, you'd get a key error. 

as a quick test, you can run `{}["key"]` and `{}.get("key")`. the first will error, the second will return None. so you can chain a bunch of .get(key, {}) together to traverse a nested dictionary.

Reply:
Oh, that's great. And my bad, thought I knew what `get` did; should have verified before asking.